cmd/compute-domain-daemon/main.go

```diff
@@ -0,0 +1,247 @@
+		return nil
+	}
+
+	// Check if IMEX daemon is ready
+	cmd := exec.CommandContext(ctx, imexCtl, "-q", "-i", "127.0.0.1", "50005")
+	output, err := cmd.Output()
+	if err != nil {
+		return fmt.Errorf("error checking IMEX daemon status: %w", err)
+	}
+
+	if string(output) != "READY\n" {
+		return fmt.Errorf("IMEX daemon not ready: %s", string(output))
+	}
+
+	return nil
+}
+
+// checkIMEXCapable checks if the node is capable of running IMEX by verifying
+// that all GPUs on the node have consistent ClusterUUID and CliqueID values.
+// It returns true if the node is IMEX capable, false otherwise.
+func checkIMEXCapable(ctx context.Context) (bool, error) {
+	lib, err := newDeviceLib()
+	if err != nil {
+		return false, fmt.Errorf("error creating device library: %w", err)
+	}
+
+	if err := lib.init(); err != nil {
+		return false, fmt.Errorf("error initializing device library: %w", err)
+	}
+	defer lib.alwaysShutdown()
+
+	uniqueClusterUUIDs := make(map[string]struct{})
+	uniqueCliqueIDs := make(map[string]struct{})
+
+	err = lib.VisitDevices(func(i int, d nvdev.Device) error {
+		isFabricAttached, err := d.IsFabricAttached()
+		if err != nil {
+			return fmt.Errorf("error checking if device is fabric attached: %w", err)
+		}
+		if !isFabricAttached {
+			return nil
+		}
+
+		info, ret := d.GetGpuFabricInfo()
+		if ret != nvml.SUCCESS {
+			return fmt.Errorf("failed to get GPU fabric info: %w", ret)
+		}
+
+		clusterUUID, err := uuid.FromBytes(info.ClusterUuid[:])
+		if err != nil {
+			return fmt.Errorf("invalid cluster UUID: %w", err)
+		}
+
+		cliqueID := fmt.Sprintf("%d", info.CliqueId)
+
+		uniqueClusterUUIDs[clusterUUID.String()] = struct{}{}
+		uniqueCliqueIDs[cliqueID] = struct{}{}
+
+		return nil
+	})
+	if err != nil {
+		return false, fmt.Errorf("error getting fabric information from one or more devices: %w", err)
+	}
+
+	if len(uniqueClusterUUIDs) == 0 && len(uniqueCliqueIDs) == 0 {
+		return false, nil
+	}
+
+	if len(uniqueClusterUUIDs) != 1 {
+		return false, fmt.Errorf("unexpected number of unique ClusterUUIDs found on devices")
+	}
+
+	if len(uniqueCliqueIDs) != 1 {
+		return false, fmt.Errorf("unexpected number of unique CliqueIDs found on devices")
+	}
+
+	return true, nil
+}
+
+// printNodesConfig reads and prints the contents of the nodes configuration file.
+// It returns an error if the file cannot be read.
+func printNodesConfig(ctx context.Context) error {
+	fmt.Printf("%s:\n", nodesConfig)
+	content, err := os.ReadFile(nodesConfig)
+	if err != nil {
+		return fmt.Errorf("failed to read nodes config: %w", err)
+	}
+	fmt.Println(string(content))
+	return nil
+}
+
+// runIMEXDaemon starts the IMEX daemon with the specified configuration file.
+// It returns an error if the daemon fails to start or exits unexpectedly.
+func runIMEXDaemon(ctx context.Context, config string) error {
+	cmd := exec.CommandContext(ctx, imexBinary, "-c", config)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
```

Comment:
I forgot some details of what we talked about. But now when we wrap the IMEX daemon process like this, let's review some config settings. Maybe:

- `LOG_FILE_NAME=""` -- according to docs, if empty, this emits to stderr
- `DAEMONIZE=0` -- maybe that's even required to to properly shut it down?

We currently also still have `IMEX_WAIT_FOR_QUORUM=RECOVERY`

(https://github.com/NVIDIA/k8s-dra-driver-gpu/blob/9732ac32c1e77dc9b08cf741830b5fb3529787d4/templates/compute-domain-daemon-config.tmpl.cfg#L106C1-L106C30)

Whereas I think in our conversations we explored setting this to

> NONE: Do not wait for any quorum with other nodes.